### DISCUSSION TITLE
Return the time random connection strategy

### DISCUSSION THREAD

**Discussion by larry-cdn77** [ClickHouse/clickhouse-go#1302] [Feature requests]
It seems the time random strategy did not make it through the v2 refactoring

Randomisation is useful to prevent herding in distributed applications (simultaneous restart of multiple instances of a service)

If it makes sense to reintroduce I can have a go and raise a pull request

**Comment by jkaflik** [comment]
Hi @larry-cdn77
could you clarify what do you mean by "time random connection"? Does it mean to accquite random connection from connection pool?

**Reply by larry-cdn77** [reply]
Thank you for responding

I was referring to d276600 (#452) which moved the implementation of connection strategies from connect.go to clickhouse.go

It also reduced them from

- random
- in-order
- time random

to

- round robin
- in-order

It is 'time random' that I find useful for load balancing

**Reply by jkaflik** [reply]
@larry-cdn77 clear. Feel free to submit a PR with a change.
